src/main/java/com/intuit/graphql/authorization/enforcement/RedactingVisitor.java

```diff
@@ -41,18 +43,20 @@ public void visitField(QueryVisitorFieldEnvironment queryVisitorFieldEnvironment
 
     //first check if the type itself is permitted
     //this is just an optimization to keep from descending on each field node
-    boolean permitted = typeFieldPermissionVerifier.isPermitted(requestedFieldDefinition.getType());
+    boolean permitted = typeFieldPermissionVerifier.isPermitted(unwrapAll(requestedFieldDefinition.getType()));
```

Comment:
This is expecting the list Type to be present in the Map. So if you run into [Book] and Book inside the map which is not desirable.